### DISCUSSION TITLE
Dictionaries now sorted by steno order rather than alphabetically?

### DISCUSSION THREAD

**Discussion by pete-debiase** [opensteno/plover#1454] [Technical Discussion]
Dear @benoit-pierre and @morinted,

Howdy 👋 ! Long-time Plover user here, first time starting to really dig around in the source code.

I was wondering if you could help me understand the goal of 3d57e9af? As someone who has historically spent a lot of time editing dictionaries by hand in a text editor (for batch-editing purposes, etc.), I found the switch from sorting alphabetically to sorting by steno order to be rather unexpected (and jarring).

One possible benefit of this commit (I think?) is that it causes dictionary entries to be sorted by hand shape rather than the spelling of the outline. This sorting order may also potentially be more common in conventional steno technologies (but totally don't know).

In practice, I have found that sorting alphabetically already tends to group entries by hand shape, but this may just be an artifact of my own personal steno dictionaries - I tend not to add all the different possible misstrokes of an outline (just one "correct" entry per word/lemma).

Also, I realize this is very much a case of [every change breaks someone's workflow](https://xkcd.com/1172/) :) , but would welcome any thoughts you may be able to share.

Thanks as always for making Plover awesome!

**Comment by user202729** [comment]
It was a feature request. https://github.com/openstenoproject/plover/issues/1017 ... so go ask that user?

There remains the option of editing the entries *only* by hand (or make a hand-edit-only one and a Plover-edit-only one), or make some external script to re-sort etc.

**Comment by panathea** [comment]
The dictionary was previously sorted alphabetically because it had to be sorted somehow, and alphabetical is the default for strings.

Steno _does_ have an order, though: its own alphabet. It makes a lot more sense for us to sort dictionaries by steno order because… well because it's steno and not words.

The implementation is cross-system and will apply to other languages and systems, too. In English steno, right-hand-only chords start with `-` which makes sorting somewhat natural, but other steno systems may not use the hyphen at all.

Maybe you can share what you find jarring about this change and how it affects your ability to edit the dictionaries by hand?

**Comment by pete-debiase** [comment]
That makes perfect sense. I was unaware that this was part of a larger feature request and that it gelled with everyone so well, which is why I asked.

> Maybe you can share what you find jarring about this change and how it affects your ability to edit the dictionaries by hand?

Like all steno users, I have just spent tons of time over the last X years on my dictionaries - I had gotten very used to Plover operating on them in a certain way and working on them by hand in a certain way. It's totally not a big deal - I can just get used to the new way, or make a local change on a personal fork, or maybe try submitting a PR to make it user-configurable.

As @user202729 suggested, I already keep dictionaries in which sort order is especially important away from the top of the dictionary stack and only edit them by hand (for keyboard shortcuts and other command-like entries that are more maintainable when grouped by type).

Anyways, apologies if I was barking up the wrong tree on this one, and thanks so much for the discussion and helping me learn :) !

**Reply by panathea** [reply]
No worries at all—please always feel free to ask questions.

When I edit dictionaries, I usually don't worry about the order of the entries and then modify something in Plover so that it re-sorts them.

One nice recent change is that on dictionary load, invalid entries are highlighted in a notification 🎉 which I just love.
